### PR TITLE
[knex] Multi-column whereIn clauses with subqueries and QueryCallbacks

### PR DESCRIPTION
**Pull request by scotttrinh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://knexjs.org/#Builder-whereIn
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/knex/index.d.ts [hunk 1]
```diff
@@ -307,6 +307,11 @@ declare namespace Knex {
 
     interface WhereIn {
         (columnName: string, values: Value[]): QueryBuilder;
+        (columnNames: string[], values: Value[][]): QueryBuilder;
+        (columnName: string, subquery: QueryBuilder): QueryBuilder;
+        (columnNames: string[], subquery: QueryBuilder): QueryBuilder;
+        (columnName: string, callback: QueryCallback): QueryBuilder;
+        (columnNames: string[], callback: QueryCallback): QueryBuilder;
```
Comment (mastermatt) [inline types/knex/index.d.ts]: Could this also be expressed like this?
`(columnNames: string|string[], callback: Value[]|Value[][]|QueryBuilder|QueryCallback): QueryBuilder;`

I'm not sure which way is preferred.

Comment (scotttrinh) [reply types/knex/index.d.ts]: How about:
```typescript
(columnName: string, value: Value[] | QueryBuilder | QueryCallback): QueryBuilder
(columnNames: string[], values: Value[][] | QueryBuilder | QueryCallback): QueryBuilder
```

Comment (scotttrinh) [reply types/knex/index.d.ts]: ⚡️

**Comment by scotttrinh** [comment]
Also, looks like there might be a bunch of duplicate `interface`s declared around here. I'm surprised that the linter or compiler doesn't complain about that sort of thing. Maybe I'm missing if there is a point to declaring the same interface multiple times? I went ahead and removed the duplicate for `WhereIn`.

**Review by mastermatt — APPROVED** [review]

**Pull request merged**